File: app/models/state.rb
# app/models/state.rb
# Copyright 2014 Sergei Ianovich
#
# Licensed under AGPLv3 or later
# Process Control Service Web Interface

class State
  include Mongoid::Document
  include Mongoid::Timestamps::Created::Short
  include MongoidPaginator

  belongs_to :device

  field :content, type: String
  field :stamp,   type: DateTime
  field :start,   type: Integer
  field :size,    type: Integer

  index({ device_id: 1, c_at: -1 })

  def time
    if (DateTime.now - stamp) > 1
      stamp.strftime('%H:%M')
    else
      stamp.strftime('%-d %b')
    end
  end
end

# vim:ts=2 sts=2 sw=2 et:

File: models/state.js
/* models/state.js
 * Copyright 2014 Sergei Ianovich
 *
 * Licensed under AGPL-3.0 or later, see LICENSE
 * Process Control Service Web Interface
 */

var mongoose = require('mongoose');
var Device = require('./device');

var state_schema = new mongoose.Schema({
  device: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Device'
  },
  stamp: Date,
  outputs: {}
});

state_schema.index({ device: 1, stamp: -1, _id: 1 });

var State = mongoose.model('State', state_schema);

module.exports = State;

// vim:ts=2 sts=2 sw=2 et:
